docs/API-functions.md

```diff
@@ -1614,6 +1618,284 @@ is not availble - it then sets the value to `SRT_REJC_PREDEFINED + 404`.
+to check call parameters.
+
+#### `SRT_EDUPLISTEN`
+
+The port tried to be bound for listening is already busy. Note that binding
+to the same port is allowed in general (when `SRTO_REUSEADDR` is true on
+every socket that bound it), but only one such socket can be a listener.
+
+#### `SRT_ELARGEMSG`
+
+Size exceeded. This is reported in the following situations:
+
+* Trying to receive a message, but the read-ready message is larger than
+the buffer passed to the receiving function
+
+* Trying to send a message, but the size of this message exceeds the
+size of the preset sender buffer, so it cannot be stored in the sender buffer.
+
+* With getting group data, the array to be filled is too small.
+
+
+#### `SRT_EINVPOLLID`
+
+The epoll ID passed to an epoll function is invalid
+
+#### `SRT_EPOLLEMPTY`
+
+The epoll container has currently no subscribed sockets. This is reported by an
+epoll waiting function that would in this case block forever. This problem
+might be reported both in a situation that you have created a new epoll
+container and didn't subscribe any socket to it, or you did, but all these
+sockets have been closed (including when closed in a separate thread while the
+waiting function was blocking). Note that this situation can be prevented from
+by setting the `SRT_EPOLL_ENABLE_EMPTY` flag, which may be useful in case when
+you use multiple threads and start waiting without subscribed sockets, so that
+you can subscribe them later from another thread.
+
+#### `SRT_EASYNCFAIL`
+
+General asynchronous failure (not in use currently).
+
+
+#### `SRT_EASYNCSND`
+
+Sending operation is not ready to perform. This error is reported
+when trying to perform a sending operation on a socket that is not
+ready for sending, but `SRTO_SNDSYN` was set to false (when true,
+the function would block the call otherwise).
+
+#### `SRT_EASYNCRCV`
+
+Receiving operation is not ready to perform. This error is reported
+when trying to perform a receiving operation or accept a new socket from the
+listner socket, when the socket is not ready for that operation, but
+`SRTO_RCVSYN` was set to false (when true, the function would block
+the call otherwise).
+
+#### `SRT_ETIMEOUT`
+
+The operation timed out. This can happen if you have a timeout
+set by an option (`SRTO_RCVTIMEO` or `SRTO_SNDTIMEO`), or passed
+as an extra argument (`srt_epoll_wait` or `srt_accept_bond`) and
+the function call was blocking, but the required timeout time has passed.
+
+#### `SRT_ECONGEST`
+
+This error is used only in an experimental version that requires
+setting the `SRT_ENABLE_ECN` macro at compile time. Otherwise the
+described below situation results in just usual successful report.
+
+This error should be reported by the sending function in case when
+with `SRTO_TLPKTDROP` set to true there were packets had to be removed
+from the sender buffer and forgotten because the network is unable
+to live up to the sending rate. 
```

Comment:
```suggestion
This error should be reported by the sending function when, with
`SRTO_TLPKTDROP` set to true, packets to be removed from the 
sender buffer are forgotten because the network is unable
to support to the sending rate. 
```

Reply:
Ok, I'll try to describe the situation. When this error is encountered:
 - SRTO_TLPKTDROP flag was set
 - the TLPKTDROP on the sender site was triggered (there were found packets in the sender buffer, which's predicted delivery time is already in the past, and they are still not acknowledged - in that case the sender simply drops these packets from the sender buffer and they are no longer eligible for retransmission)

And also the `SRT_ENABLE_ECN` was turned on at compile time.

In this case the data sending function will return this error, however the data that it has scheduled for sending have been accepted.

Maybe this way:

This error should be reported by the sending function when, with `SRTO_TSBPDMODE` and `SRTO_TLPKTDROP` set to true`, some packets were dropped at the sender side (see description for `SRTO_TLPKTDROP` for details). This doesn't concern the data that were passed for sending by this call (these data are placed at the back side of the sender buffer, while the dropped packets are at the front side of the sender buffer). In other words, the operation done by the sending function is actually successful, just the application might want to slowdown the sending rate to avoid the congestion.

Reply:
Your last suggestion is much better. Some minor tweaks:

This error should be reported by the sending function when, with `SRTO_TSBPDMODE` and `SRTO_TLPKTDROP` set to true, some packets were dropped at the sender side (see the description of `SRTO_TLPKTDROP` for details). This doesn't concern the data that were passed for sending by the sending function (these data are placed at the back of the sender buffer, while the dropped packets are at the front). In other words, the operation done by the sending function is successful, but the application might want to slow down the sending rate to avoid congestion.